server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -240,6 +271,16 @@ public Task unregister(Task task) {
         // Decrement the task's self-thread as part of unregistration.
         task.decrementResourceTrackingThreads();
 
+        if (taskResourceConsumersEnabled && task.getResourceStats().isEmpty() == false) {
```

Comment:
No need for the empty check because the for loop will be a no-op for an empty set

Reply:
Thats right. Removed the check.